pkg/tsdb/jaeger/querydata.go

```diff
@@ -50,7 +75,66 @@ func queryData(ctx context.Context, dsInfo *datasourceInfo, req *backend.QueryDa
 	return response, nil
 }
 
-// transformTraceResponse converts Jaeger trace data to a Data frame
+func transformSearchResponse(response []TraceResponse, dsInfo *datasourceInfo) *data.Frame {
+	frame := data.NewFrame("traces",
+		data.NewField("traceID", nil, []string{}).SetConfig(&data.FieldConfig{
+			DisplayName: "Trace ID",
+			Links: []data.DataLink{
+				{
+					Title: "Trace: ${__value.raw}",
+					URL:   "",
+					Internal: &data.InternalDataLink{
+						DatasourceUID:  dsInfo.JaegerClient.settings.UID,
+						DatasourceName: dsInfo.JaegerClient.settings.Name,
+						Query: map[string]interface{}{
+							"query": "${__value.raw}",
+						},
+					},
+				},
+			},
+		}),
+		data.NewField("traceName", nil, []string{}).SetConfig(&data.FieldConfig{
+			DisplayName: "Trace name",
+		}),
+		data.NewField("startTime", nil, []time.Time{}).SetConfig(&data.FieldConfig{
+			DisplayName: "Start time",
+		}),
+		data.NewField("duration", nil, []int64{}).SetConfig(&data.FieldConfig{
+			DisplayName: "Duration",
+			Unit:        "µs",
+		}),
+	)
+
+	frame.Meta = &data.FrameMeta{
+		PreferredVisualization: "table",
+	}
+
+	for _, trace := range response {
+		if len(trace.Spans) == 0 {
+			continue
+		}
+
+		rootSpan := trace.Spans[0]
```

Comment:
Is this root span for sure? Because on frontend we have some extensive logic, so could you provide more context on this.

Reply:
for this case the rootSpan is needed to find the start time of the trace, so its safe to assume the root is just the span with the earliest startTime. updated in https://github.com/grafana/grafana/pull/103399/commits/7b9a5851b6155d4db45af4de048aae903e7e3de8 to reflect this.